### ISSUE TITLE
CogVideoX1.5-5B-I2 finetuning unexpected results

### ISSUE THREAD

**Issue by ZukaChilachava** [a-r-r-o-w/finetrainers#160]
### System Info / 系統信息

Cuda: 12.2
Diffusers: 0.33.0.dev0
Python: 3.10
OS: Linux
Torch: 2.4.1

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

Hello, tried finetuning the CogVideoX1.5-5B-I2 model and the resulting videos from inference just seem to turn the video saturation up, nothing more. 0 movement.

prepare_dataset.sh
```bash
#!/bin/bash

MODEL_ID="THUDM/CogVideoX1.5-5B-I2V"

NUM_GPUS=1

# For more details on the expected data format, please refer to the README.
DATA_ROOT="/root/finetrainers/train_data"  # This needs to be the path to the base directory where your videos are located.
CAPTION_COLUMN="prompts.txt"
VIDEO_COLUMN="videos.txt"
OUTPUT_DIR="/root/finetrainers/datasets/preprocessed-dataset"
HEIGHT_BUCKETS="480"
WIDTH_BUCKETS="720"
FRAME_BUCKETS="5"
MAX_NUM_FRAMES="5"
MAX_SEQUENCE_LENGTH=226
TARGET_FPS=5
BATCH_SIZE=1
DTYPE=fp16

# To create a folder-style dataset structure without pre-encoding videos and captions
# For Image-to-Video finetuning, make sure to pass `--save_image_latents`
CMD_WITHOUT_PRE_ENCODING="\
  torchrun --nproc_per_node=$NUM_GPUS \
    training/cogvideox/prepare_dataset.py \
      --model_id $MODEL_ID \
      --data_root $DATA_ROOT \
      --caption_column $CAPTION_COLUMN \
      --video_column $VIDEO_COLUMN \
      --output_dir $OUTPUT_DIR \
      --height_buckets $HEIGHT_BUCKETS \
      --width_buckets $WIDTH_BUCKETS \
      --frame_buckets $FRAME_BUCKETS \
      --max_num_frames $MAX_NUM_FRAMES \
      --max_sequence_length $MAX_SEQUENCE_LENGTH \
      --target_fps $TARGET_FPS \
      --batch_size $BATCH_SIZE \
      --dtype $DTYPE \
      --save_image_latents
"

CMD_WITH_PRE_ENCODING="$CMD_WITHOUT_PRE_ENCODING --save_latents_and_embeddings"

# Select which you'd like to run
CMD=$CMD_WITH_PRE_ENCODING

echo "===== Running \`$CMD\` ====="
eval $CMD
echo -ne "===== Finished running script =====\n"
```

train_image_to_video_lora.sh
```bash
export TORCH_LOGS="+dynamo,recompiles,graph_breaks"
export TORCHDYNAMO_VERBOSE=1
export WANDB_MODE="offline"
export NCCL_P2P_DISABLE=1
export TORCH_NCCL_ENABLE_MONITORING=0

GPU_IDS="0"

LEARNING_RATES=("1e-4" "1e-3")
LR_SCHEDULES=("cosine_with_restarts")
OPTIMIZERS=("adamw" "adam")
MAX_TRAIN_STEPS=("5000")

# Single GPU uncompiled training
ACCELERATE_CONFIG_FILE="accelerate_configs/uncompiled_1.yaml"

DATA_ROOT="/root/finetrainers/datasets/preprocessed-dataset"
CAPTION_COLUMN="prompts.txt"
VIDEO_COLUMN="videos.txt"

# Launch experiments with different hyperparameters
for learning_rate in "${LEARNING_RATES[@]}"; do
  for lr_schedule in "${LR_SCHEDULES[@]}"; do
    for optimizer in "${OPTIMIZERS[@]}"; do
      for steps in "${MAX_TRAIN_STEPS[@]}"; do
        output_dir="/root/finetrainers/model_checkpoints/cogvideox-lora__optimizer_${optimizer}__steps_${steps}__lr-schedule_${lr_schedule}__learning-rate_${learning_rate}/"

        cmd="accelerate launch --config_file $ACCELERATE_CONFIG_FILE --gpu_ids $GPU_IDS training/cogvideox/cogvideox_image_to_video_lora.py \
          --pretrained_model_name_or_path THUDM/CogVideoX1.5-5B-I2V \
          --data_root $DATA_ROOT \
          --caption_column $CAPTION_COLUMN \
          --video_column $VIDEO_COLUMN \
          --load_tensors \
          --height_buckets 480 \
          --width_buckets 720 \
          --frame_buckets 5 \
          --dataloader_num_workers 8 \
          --pin_memory \
          --seed 42 \
          --rank 128 \
          --lora_alpha 128 \
          --mixed_precision bf16 \
          --output_dir $output_dir \
          --max_num_frames 5 \
          --train_batch_size 1 \
          --max_train_steps $steps \
          --checkpointing_steps 1000 \
          --gradient_accumulation_steps 4 \
          --gradient_checkpointing \
          --learning_rate $learning_rate \
          --lr_scheduler $lr_schedule \
          --lr_warmup_steps 200 \
          --lr_num_cycles 1 \
          --enable_slicing \
          --enable_tiling \
          --noised_image_dropout 0.05 \
          --optimizer $optimizer \
          --beta1 0.9 \
          --beta2 0.95 \
          --weight_decay 0.001 \
          --max_grad_norm 1.0 \
          --allow_tf32 \
          --report_to wandb \
          --nccl_timeout 1800"
        
        echo "Running command: $cmd"
        eval $cmd
        echo -ne "-------------------- Finished executing script --------------------\n\n"
      done
    done
  done
done
```

modified finetrainers/training/cogvideox/dataset.py from line 126
the back calculation of frames for  frame_count = 5 was working incorrectly. 5/2=2, 2*4 = 8. so I added these lines to fix that
```python
latent_num_frames = video_latents.size(1)
if latent_num_frames != 2 and latent_num_frames % 2 == 0 or latent_num_frames == 1:
      num_frames = latent_num_frames * 4
else:
      num_frames = (latent_num_frames - 1) * 4 + 1
```
The video files in my dataset are 1 second long with 5 frames

this is what the loss curve looks like:
![W B Chart 12_28_2024, 9_12_06 PM](https://github.com/user-attachments/assets/d38e5169-bb7c-44ea-85d3-bab59a8d1123)

inference script:
```python
import torch

from PIL import Image
from diffusers.utils import export_to_video
from diffusers import CogVideoXImageToVideoPipeline

pipe = CogVideoXImageToVideoPipeline.from_pretrained(
    "THUDM/CogVideoX1.5-5B-I2V", torch_dtype=torch.bfloat16
).to("cuda")
pipe.load_lora_weights("model_checkpoints/cogvideox-lora__optimizer_adamw__steps_5000__lr-schedule_cosine_with_restarts__learning-rate_1e-3/checkpoint-2000/pytorch_lora_weights.safetensors", adapter_name="cogvideox-lora")
pipe.set_adapters(["cogvideox-lora"], [1.0])

# del pipe.transformer.patch_embed.pos_embedding
pipe.transformer.patch_embed.use_learned_positional_embeddings = False
pipe.transformer.config.use_learned_positional_embeddings = False

pipe.enable_model_cpu_offload()
pipe.vae.enable_slicing()
pipe.vae.enable_tiling()

image = Image.open("initial_image.png")
video = pipe(
    prompt="some prompt", 
    image=image, 
    height=480, 
    width=720, 
    num_frames=5,
    num_inference_steps=50
).frames[0]
export_to_video(video, "output.mp4", fps=5)
```

Any help/suggestions would be greatly appreciated, maybe i'm doing something wrong or is it maybe the case that for this few frames the model breaks and isn't suited for the task i'm trying to accomplish? not sure.

### Expected behavior / 期待表现

The output videos to have some motion and look like the finetuning worked.

**Comment by cdfan0627** [comment]
May I ask how you finally solved this problem?
